react/TextField/TextField.js

```diff
@@ -88,11 +88,12 @@ export default class TextField extends Component {
   }
 
   renderInput() {
-    const { inputProps = {}, id } = this.props;
+    const { inputProps = {}, id, valid } = this.props;
     const { ref } = inputProps;
     const allInputProps = {
       ...combineClassNames(inputProps, styles.input),
       ...(id ? { id } : {}),
+      ...{ 'aria-invalid': valid === false },
```

Comment:
I just tested it and the spread ensures that if you pass in an `aria-invalid`, the prop inside the component will take precedence.